src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.Bindings;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding.Http
+{
+    internal class HttpTriggerAttributeBindingProvider : ITriggerBindingProvider
+    {
+        public HttpTriggerAttributeBindingProvider()
+        {
+        }
+
+        public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            ParameterInfo parameter = context.Parameter;
+            HttpTriggerAttribute attribute = parameter.GetCustomAttribute<HttpTriggerAttribute>(inherit: false);
+            if (attribute == null)
+            {
+                return Task.FromResult<ITriggerBinding>(null);
+            }
+
+            // Can bind to user types, HttpRequestMessage and all the Read
+            // Types supported by StreamValueBinder
+            IEnumerable<Type> supportedTypes = StreamValueBinder.GetSupportedTypes(FileAccess.Read)
+                .Union(new Type[] { typeof(HttpRequestMessage) });
+            bool isSupportedTypeBinding = ValueBinder.MatchParameterType(parameter, supportedTypes);
+            bool isUserTypeBinding = !isSupportedTypeBinding && parameter.ParameterType.IsClass;
```

Comment:
I was wondering if we really want to exclude value types here. If excluding structs is not a requirement, it would be nice to relax that restriction. We should be able to handle struct creation/deserialization.